src/GradientPicker/index.js

```diff
@@ -0,0 +1,1458 @@
+                this.gField.value,
+                this.bField.value,
+                this.aField.value
+            ].map(function (v) {
+                return v / 255;
+            });
+            this.hsva = this.Helpers.toHsva(rgba);
+            this.colorSelectedAnchor(this.color);
+        }
+    }
+
+    _onHexChanged() {
+        if (!this._changing) {
+            const hex = this.hexField.value.trim().toLowerCase();
+            if (/^([0-9a-f]{2}){3,4}$/.test(hex)) {
+                const rgb = [parseInt(hex.substring(0, 2), 16),
+                    parseInt(hex.substring(2, 4), 16),
+                    parseInt(hex.substring(4, 6), 16)];
+                this.hsva = this._rgb2hsv(rgb).concat([this.hsva[3]]);
+                this.colorSelectedAnchor(this.color);
+            }
+        }
+    }
+
+    _onMouseDown(evt) {
+        if (evt.currentTarget === this.colorRect.element) {
+            this._dragMode = 1;     // drag color
+        } else if (evt.currentTarget === this.hueRect.element) {
+            this._dragMode = 2;     // drag hue
+        } else {
+            this._dragMode = 3;     // drag alpha
+        }
+
+        this._storeHsva = this._hsva.slice();
+        this._onMouseMove(evt);
+
+        // hook up mouse
+        window.addEventListener('mousemove', this.moveHandler);
+        window.addEventListener('mouseup', this.upHandler);
+    }
+
+    _onMouseMove(evt) {
+        let newhsva;
+        if (this._dragMode === 1) {
+            const m = this.Helpers.normalizedCoord(this.colorRect, evt.pageX, evt.pageY);
+            const s = math.clamp(m[0], 0, 1);
+            const v = math.clamp(m[1], 0, 1);
+            newhsva = [this.hsva[0], s, 1 - v, this._hsva[3]];
+        } else if (this._dragMode === 2) {
+            const m = this.Helpers.normalizedCoord(this.hueRect, evt.pageX, evt.pageY);
+            const h = math.clamp(m[1], 0, 1);
+            newhsva = [h, this.hsva[1], this.hsva[2], this.hsva[3]];
+        } else {
+            const m = this.Helpers.normalizedCoord(this.alphaRect, evt.pageX, evt.pageY);
+            const a = math.clamp(m[1], 0, 1);
+            newhsva = [this.hsva[0], this.hsva[1], this.hsva[2], 1 - a];
+        }
+        if (newhsva[0] !== this._hsva[0] ||
+            newhsva[1] !== this._hsva[1] ||
+            newhsva[2] !== this._hsva[2] ||
+            newhsva[3] !== this._hsva[3]) {
+            this.hsva = newhsva;
+            this.emit('changing', this.color);
+        }
+    }
+
+    _onMouseUp(evt) {
+        window.removeEventListener('mousemove', this.moveHandler);
+        window.removeEventListener('mouseup', this.upHandler);
+
+        if (this._storeHsva[0] !== this._hsva[0] ||
+            this._storeHsva[1] !== this._hsva[1] ||
+            this._storeHsva[2] !== this._hsva[2] ||
+            this._storeHsva[3] !== this._hsva[3]) {
+            this.colorSelectedAnchor(this.color);
+        }
+    }
+
+    _hsv2rgb(hsv) {
```

Comment:
please move the two implementations of hsv / rgb conversion into its own module

Reply:
Moved to /Math/color-value.js

Reply:
please also remove from here :)